doc/rst/language/spec/types.rst

```diff
@@ -589,3 +589,156 @@ generic (:ref:`Type_Aliases_in_Generic_Types`).
 
       2
 
+
+.. _Querying_the_Type_of_an_Expression:
+
+Querying the Type of an Expression
+----------------------------------
+
+The type of a an expression can be queried with ``.type``. This
+functionality is particularly useful when doing generic programming
+(see :ref:`Chapter-Generics`).
+
+   *Example (dot-type.chpl)*.
+
+   For example, this code uses ``.type`` to query the type of the
+   variable ``x`` and store that in the type alias ``t``:
+
+   .. code-block:: chapel
+
+      var x: int;
+      type t = x.type;
+
+   .. BLOCK-test-chapelpost
+
+      writeln(t:string);
+
+   .. BLOCK-test-chapeloutput
+
+      int(64)
+
+   *Open issue*.
+
+   For types that include a runtime component, it would be useful to be
+   able to query the static and runtime components separately.
+
+
+.. _Operators_Available_on_Types:
+
+Operators Available on Types
+----------------------------
+
+There are several operators that can operate on type aliases or type
+expressions (such as the ``.type`` queries described above).
+
+Types can also be passed to a ``type`` formal of a generic function (see
+:ref:`Formal_Type_Arguments`).
+
```

Comment:
Let us remove "also" and "additionally" throughout this section. They do not improve the presentation.